### PR TITLE
Switch to //lib/tonic's DartConverter

### PR DESCRIPTION
**Pull request by abarth**
Also, switch over to //lib/tonic's typed data classes.

### CODE REVIEW

**Comment by abarth** [comment]
@Hixie

File: flutter/lib/jni/dart_jni.cc [hunk 1]
```diff
@@ -188,25 +190,38 @@ void DartJni::InitForGlobal(DartJniIsolateDataProvider provider) {
...
+                                     DART_REGISTER_NATIVE_STATIC(JniLongArray,
+                                                                 Create)
+                                         DART_REGISTER_NATIVE_STATIC(
+                                             JniObjectArray, Create)
+                                             DART_REGISTER_NATIVE_STATIC(
+                                                 JniShortArray, Create)
+                                                 DART_REGISTER_NATIVE_STATIC(
+                                                     JniString, Create)
+                                                     FOR_EACH_BINDING(
+                                                         DART_REGISTER_NATIVE)});
```
Comment (Hixie) [inline flutter/lib/jni/dart_jni.cc]: your formatter failed you

File: sky/engine/core/script/dart_init.cc [hunk 2]
```diff
@@ -90,22 +92,20 @@ static const char* kDartProfilingArgs[] = {
...
-  "--background_compilation",
+    "--background_compilation",
 };
 
 static const char* kDartCheckedModeArgs[] = {
-    "--enable_asserts",
-    "--enable_type_checks",
-    "--error_on_bad_type",
+    "--enable_asserts", "--enable_type_checks", "--error_on_bad_type",
     "--error_on_bad_override",
```
Comment (Hixie) [inline sky/engine/core/script/dart_init.cc]: your formatter failed you here too

**Comment by Hixie** [comment]
There's a bunch of places where the formatter made suboptimal choices (not as bad as the two above, but still ugly), but other than that, RSLGTM.

**Pull request merged**